### PR TITLE
Check hypergraphs

### PR DESCRIPTION
**Pull request by Coerulatus**
#247 
Changed the hypergraph models to use the Conv class defined in this library. Due to how the attention is calculated it was not always feasible to use the Conv operation. Hypergat and hypersage still use their costume convolution functions.

### CODE REVIEW

File: test/nn/hypergraph/test_unigin_layer.py [hunk 1]
```diff
@@ -15,7 +15,13 @@ def UniGIN_layer(self):
         self.in_channels = 10
         return UniGINLayer(in_channels=self.in_channels)
 
-    def test_forward(self, UniGIN_layer):
+    @pytest.fixture
+    def UniGIN_layer2(self):
```
Comment (ninamiolane) [inline test/nn/hypergraph/test_unigin_layer.py:19]: Avoid mix of under_scores and CamelCase in names.
Python guidelines ask for all lower cases and under_scores in function names, see Pep8. Can you adapt the naming of this function and the others which may have the same problem? Thanks.

File: test/nn/hypergraph/test_unigin_layer.py [hunk 2]
```diff
@@ -15,7 +15,13 @@ def UniGIN_layer(self):
...
+        """Return a UniGIN layer."""
+        self.in_channels = 10
+        return UniGINLayer(in_channels=self.in_channels, use_norm=True)
+
+    def test_forward(self, UniGIN_layer, UniGIN_layer2):
```
Comment (ninamiolane) [inline test/nn/hypergraph/test_unigin_layer.py:24]: No mix of under_scores and CamelCase.
If you're passing a class --> CamelCase
If you're passing an object or a function --> underscores.
See pep8 guidelines for international standards in code style.

File: topomodelx/nn/hypergraph/hypergat.py [hunk 3]
```diff
@@ -42,6 +45,7 @@ def __init__(
                 )
             )
         self.layers = torch.nn.ModuleList(layers)
+        self.layer_drop = torch.nn.Dropout(layer_drop)
```
Comment (ninamiolane) [inline topomodelx/nn/hypergraph/hypergat.py:48]: 👍

File: topomodelx/nn/hypergraph/hypersage_layer.py [hunk 4]
```diff
@@ -51,6 +51,8 @@ class HyperSAGELayer(MessagePassing):
         Dimension of the input features.
     out_channels : int
         Dimension of the output features.
+    alpha: int, default=-1
```
Comment (ninamiolane) [inline topomodelx/nn/hypergraph/hypersage_layer.py:54]: NIT: there is a space before the ":" in numpy's docstring conventions (which are the ones we use in the library) for some reason.

**Review by ninamiolane — APPROVED** [review]
Very nice! One super minor comment about coding style (see PEP8 and the standards on naming functions: https://peps.python.org/pep-0008/.

Once this is fixed, we can merge. Congrats on this fantastic contribution (and sorry for the delay in reviewing).

**Comment by Coerulatus** [comment]
Thank you for the feedback! I resolved the mentioned issues

**Comment by Coerulatus** [comment]
Hello,
I noticed that the pull request has not been accepted yet, and I was wondering if there might be any issues or concerns that need to be addressed.
Thank you very much for your time and consideration.

**Comment by ffl096** [comment]
This pull request has conflicts with the main branch that must be resolved before merging. Please rebase or merge the current main branch and resolve these conflicts :)

**Comment by Coerulatus** [comment]
@ffl096 Hi Florian, I have resolved the conflicts. Let me know if there is anything else! Thank you very much.

**Review by ffl096 — APPROVED** [review]

**Pull request merged**